sanic/handlers.py

```diff
@@ -27,24 +37,92 @@ class ErrorHandler:
+        self,
+        fallback: Union[str, Default] = _default,
+        base: Type[BaseRenderer] = HTMLRenderer,
     ):
         self.handlers: List[Tuple[Type[BaseException], RouteHandler]] = []
         self.cached_handlers: Dict[
             Tuple[Type[BaseException], Optional[str]], Optional[RouteHandler]
         ] = {}
         self.debug = False
-        self.fallback = fallback
+        self._fallback = fallback
         self.base = base
 
+        if fallback is not _default:
+            self._warn_fallback_deprecation()
+
+    @property
+    def fallback(self):
+        # This id for backwards compat and can be removed in v22.6
+        if self._fallback is _default:
+            return DEFAULT_FORMAT
+        return self._fallback
+
+    @fallback.setter
+    def fallback(self, value: str):
+        self._warn_fallback_deprecation()
+        if not isinstance(value, str):
+            raise SanicException(
+                f"Cannot set error handler fallback to: {value=}"
+            )
+        self._fallback = value
+
+    @staticmethod
+    def _warn_fallback_deprecation():
+        error_logger.warning(
+            DeprecationWarning(
+                "Setting the ErrorHandler fallback value directly is "
+                "deprecated and no longer supported. This feature will "
+                "be removed in v22.6. Instead, use "
+                "app.config.FALLBACK_ERROR_FORMAT."
+            ),
+        )
+
+    @classmethod
+    def _get_fallback_value(cls, error_handler: ErrorHandler, config: Config):
+        if error_handler._fallback is not _default:
+            if config._FALLBACK_ERROR_FORMAT is _default:
+                return error_handler.fallback
+
+            error_logger.warning(
+                "Conflicting error fallback values were found in the "
+                "error handler and in the app.config while handling an "
+                "exception. Using the value from app.config."
+            )
+        return config.FALLBACK_ERROR_FORMAT
+
     @classmethod
-    def finalize(cls, error_handler, fallback: Optional[str] = None):
+    def finalize(
+        cls,
+        error_handler: ErrorHandler,
+        fallback: Optional[str] = None,
+        config: Optional[Config] = None,
+    ):
+        if fallback:
+            error_logger.warning(
+                DeprecationWarning(
+                    "Setting the ErrorHandler fallback value via finalize() "
+                    "is deprecated and no longer supported. This feature will "
+                    "be removed in v22.6. Instead, use "
+                    "app.config.FALLBACK_ERROR_FORMAT."
+                ),
+            )
+
+        if config is None:
+            error_logger.warning(
+                DeprecationWarning(
+                    "Starting in v22.3, config will be a required argument "
+                    "for ErrorHandler.finalize()."
+                ),
+            )
```

Comment:
This shouldn't be needed. As soon as we remove `fallback` we should be able to deprecate the `finalize` method.
```suggestion
```

Reply:
We'll get there when we get there.